include/hyprlang.hpp

```diff
@@ -446,6 +447,12 @@ namespace Hyprlang {
 
         CConfigImpl* impl;
 
+        enum eGetNextLineFailure : uint8_t {
+            GETNEXTLINEFAILURE_EOF = 0,
+            GETNEXTLINEFAILURE_BACKSLASH,
+        };
+
+        std::expected<std::string, eGetNextLineFailure> getNextLine(std::istream& str, int &rawLineNum, int &lineNum);
```

Comment:
The enum and function don't belong in CConfig, IMO.

They don't use the locals at all, and only add a burden of `<expected>` on the TU.

I'd throw both of these into the cpp file only, as a local, `static` function

Reply:
Makes sense to me! Done.